.circleci/config.yml

```diff
@@ -0,0 +1,176 @@
+
+jobs:
+  build:
+    executor: builder
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - gradle-{{ checksum "build.gradle" }}
+            - gradle-
+      - run:
+          command: ./gradlew build
+      - run:
+          name: Record the Application Version
+          command: |
+            jar xvf build/libs/prison-estate*.jar BOOT-INF/classes/META-INF/build-info.properties
+            APP_VERSION=$(grep 'build.version=' BOOT-INF/classes/META-INF/build-info.properties | awk -F= '{print $2}')
+            rm BOOT-INF/classes/META-INF/build-info.properties
+
+            # make up a good version name
+            [[ "${CIRCLE_BRANCH}" == "master" ]] && echo "${APP_VERSION}" || echo "${APP_VERSION}.${CIRCLE_BRANCH}" > ./app_version.txt
+
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: gradle-{{ checksum "build.gradle" }}
+      - store_test_results:
+          path: build/test-results
+      - store_artifacts:
+          path: build/reports/tests
+      - persist_to_workspace:
+          root: .
+          paths:
+            - build/libs
+            - build.gradle
+            - app_version.txt
+
+  build_docker:
+    executor: deployer
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/app
+      - setup_remote_docker:
+          docker_layer_caching: true
+      - run:
+          name: Build container and publish to quay.io
+          command: |
+            APP_VERSION=$(cat ./app_version.txt)
+            IMAGE_NAME="quay.io/hmpps/prison-estate"
+            docker build --rm=false . \
+               --tag "$IMAGE_NAME:$APP_VERSION" \
+               --label "maintainer=info@digital.justice.gov.uk" \
+               --label "app.version=$APP_VERSION" \
+               --label "build.version=$APP_VERSION" \
+               --label "build.number=$CIRCLE_BUILD_NUM" \
+               --label "build.url=$CIRCLE_BUILD_URL" \
+               --label "build.gitref=$CIRCLE_SHA1"
+
+            if [[ "${CIRCLE_BRANCH}" == "master" ]]; then
+              # publish to quay.io as the tag
+              docker login -u $QUAYIO_USERNAME -p $QUAYIO_PASSWORD quay.io
+              docker tag $IMAGE_NAME:$APP_VERSION $IMAGE_NAME:latest
+              docker push $IMAGE_NAME:$APP_VERSION
+              docker push $IMAGE_NAME:latest
+            fi
+
+  deploy_dev:
+    executor: deployer
+    steps:
+      - deploy_to_env:
+          env: "dev"
+
+  deploy_preprod:
+    executor: deployer
+    steps:
+      - deploy_to_env:
+          env: "preprod"
+
+  deploy_prod:
+    executor: deployer
+    steps:
+      - deploy_to_env:
+          env: "prod"
+
+workflows:
+  version: 2
+  build-test-and-deploy:
+    jobs:
+      - build:
+          filters:
+            tags:
+              ignore: /.*/
+      - build_docker:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
+  scheduled:
```

Comment:
temporarily remove deployments while waiting for cloud platform to accept pull request